File: frontend/src/services/media.service.ts
/**
 * Media service for handling media-related API calls
 */

import { api } from '@/lib/api';
import type {
  Media,
  MediaImportRequest,
  MediaUploadRequest,
  MediaStatusUpdateRequest,
  MediaConsentRequest,
  MediaConsentResponse,
} from '@/types/media.types';

const PUBLIC_BASE_URL = '/public';
const AUTH_BASE_URL = '';

/**
 * Gets all media for a tournament
 * @param tournamentId - The tournament ID
 * @returns List of media
 */
export const getTournamentMedia = async (tournamentId: string): Promise<Media[]> => {
  const { data } = await api.get<Media[]>(`${PUBLIC_BASE_URL}/tournaments/${tournamentId}/media`);
  return data;
};

/**
 * Gets a media by its ID
 * @param id - The media ID
 * @returns The media
 */
export const getById = async (id: string): Promise<Media> => {
  const { data } = await api.get<Media>(`${PUBLIC_BASE_URL}/media/${id}`);
  return data;
};

/**
 * Imports media from Twitch
 * @param tournamentId - The tournament ID
 * @param request - The import request
 * @returns List of imported media
 */
export const importFromTwitch = async (
  tournamentId: string,
  request: MediaImportRequest
): Promise<Media[]> => {
  const { data } = await api.post<Media[]>(
    `${AUTH_BASE_URL}/tournaments/${tournamentId}/media/import`,
    request
  );
  return data;
};

/**
 * Uploads a new media
 * @param tournamentId - The tournament ID
 * @param request - The upload request
 * @returns The created media
 */
export const upload = async (
  tournamentId: string,
  request: MediaUploadRequest
): Promise<Media> => {
  const { data } = await api.post<Media>(
    `${AUTH_BASE_URL}/tournaments/${tournamentId}/media/upload`,
    request
  );
  return data;
};

/**
 * Updates the status of a media (approve/reject)
 * @param mediaId - The media ID
 * @param request - The status update request
 * @returns The updated media
 */
export const updateStatus = async (
  mediaId: string,
  request: MediaStatusUpdateRequest
): Promise<Media> => {
  const { data } = await api.put<Media>(`${AUTH_BASE_URL}/media/${mediaId}/status`, request);
  return data;
};

/**
 * Creates a media consent
 * @param request - The consent request
 * @returns The consent response
 */
export const createConsent = async (request: MediaConsentRequest): Promise<MediaConsentResponse> => {
  const { data } = await api.post<MediaConsentResponse>(`${AUTH_BASE_URL}/media/consent`, request);
  return data;
};

export const mediaService = {
  getTournamentMedia,
  getById,
  importFromTwitch,
  upload,
  updateStatus,
  createConsent,
};

File: backend/src/main/java/com/tft/tournament/service/MediaService.java
package com.tft.tournament.service;

import com.tft.tournament.dto.request.MediaConsentRequest;
import com.tft.tournament.dto.request.MediaImportRequest;
import com.tft.tournament.dto.request.MediaStatusUpdateRequest;
import com.tft.tournament.dto.request.MediaUploadRequest;
import com.tft.tournament.dto.response.MediaResponse;

import java.util.List;
import java.util.UUID;

/**
 * Service pour la gestion des médias.
 *
 * @author Fethi Benseddik
 * @version 1.0
 * @since 2025
 */
public interface MediaService {

    /**
     * Récupère tous les médias d'un tournoi.
     *
     * @param tournamentId identifiant du tournoi
     * @return la liste des médias
     */
    List<MediaResponse> getTournamentMedia(UUID tournamentId);

    /**
     * Importe des médias depuis Twitch.
     *
     * @param tournamentId identifiant du tournoi
     * @param request      la requête d'import
     * @param userId       identifiant de l'utilisateur qui effectue l'import
     * @return la liste des médias importés
     */
    List<MediaResponse> importFromTwitch(UUID tournamentId, MediaImportRequest request, UUID userId);

    /**
     * Upload un nouveau média.
     *
     * @param tournamentId identifiant du tournoi
     * @param request      la requête d'upload
     * @param userId       identifiant de l'utilisateur qui effectue l'upload
     * @return le média créé
     */
    MediaResponse uploadMedia(UUID tournamentId, MediaUploadRequest request, UUID userId);

    /**
     * Met à jour le statut d'un média (approve/reject).
     *
     * @param mediaId identifiant du média
     * @param request la requête de mise à jour
     * @param userId  identifiant de l'utilisateur qui effectue la mise à jour
     * @return le média mis à jour
     */
    MediaResponse updateMediaStatus(UUID mediaId, MediaStatusUpdateRequest request, UUID userId);

    /**
     * Crée un consentement média.
     *
     * @param request la requête de consentement
     * @return confirmation du consentement
     */
    MediaConsentResponse createConsent(MediaConsentRequest request);

    /**
     * Récupère un média par son identifiant.
     *
     * @param mediaId identifiant du média
     * @return le média
     */
    MediaResponse getMediaById(UUID mediaId);

    /**
     * Réponse de consentement média.
     *
     * @param id           identifiant du consentement
     * @param casterId     identifiant du caster
     * @param tournamentId identifiant du tournoi
     * @param consentMethod méthode de consentement
     * @param isActive     si le consentement est actif
     */
    record MediaConsentResponse(
            UUID id,
            UUID casterId,
            UUID tournamentId,
            String consentMethod,
            Boolean isActive
    ) {
    }
}
